Mage.Sets/src/mage/cards/t/ThranTome.java

```diff
@@ -0,0 +1,113 @@
+import mage.filter.FilterCard;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.Target;
+import mage.target.TargetCard;
+import mage.target.common.TargetOpponent;
+
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * @author arcox
+ */
+public final class ThranTome extends CardImpl {
+
+    public ThranTome(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{4}");
+
+        // Reveal the top three cards of your library. Target opponent chooses one of those cards. Put that card into your graveyard, then draw two cards.
+        Ability ability = new SimpleActivatedAbility(new ThranTomeEffect(), new ManaCostsImpl("{5}"));
+        ability.addCost(new TapSourceCost());
+        this.addAbility(ability);
+    }
+
+    public ThranTome(final ThranTome card) {
+        super(card);
+    }
+
+    @Override
+    public ThranTome copy() {
+        return new ThranTome(this);
+    }
+}
+
+class ThranTomeEffect extends OneShotEffect {
+
+    public ThranTomeEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "Reveal the top three cards of your library. Target opponent chooses one of those cards. Put that card into your graveyard, then draw two cards";
+    }
+
+    public ThranTomeEffect(final ThranTomeEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public ThranTomeEffect copy() {
+        return new ThranTomeEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        // validate source and controller exist
+        Player controller = game.getPlayer(source.getControllerId());
+        MageObject sourceObject = game.getObject(source.getSourceId());
+
+        if (sourceObject == null && controller == null) {
+            return false;
+        }
+
+        // target an opponent, if able
+        Player opponent;
+        Set<UUID> opponents = game.getOpponents(controller.getId());
+        opponents.removeIf(opp -> !game.getPlayer(opp).canBeTargetedBy(sourceObject, source.getControllerId(), game));
+
+        if (opponents.isEmpty()) {
+            return false;
+        } else {
+            if (opponents.size() == 1) {
+                opponent = game.getPlayer(opponents.iterator().next());
+            } else {
+                Target target = new TargetOpponent();
+                controller.chooseTarget(Outcome.Detriment, target, source, game);
+                opponent = game.getPlayer(target.getFirstTarget());
+            }
+        }
+
+        // reveal the cards and choose one. put it in the graveyard
+        Card cardToGraveyard;
+        Cards cards = new CardsImpl(controller.getLibrary().getTopCards(game, 3));
+        controller.revealCards(sourceObject.getIdName(), cards, game);
+
+        if (cards.size() == 1) {
+            cardToGraveyard = cards.getRandom(game);
+        } else {
+            TargetCard target = new TargetCard(1, Zone.LIBRARY, new FilterCard());
```

Comment:
There is a "TargetCardInLibrary" already.  Avatar of Growth for an example.

Reply:
Fixed, thanks for the tip